.github/workflows/daily.yml

```diff
@@ -824,3 +824,42 @@ jobs:
     - name: cluster tests
       if: true && !contains(github.event.inputs.skiptests, 'cluster')
       run: ./runtest-cluster ${{github.event.inputs.cluster_test_args}}
+
+  test-ubuntu-compile-time-detection:
+    runs-on: ubuntu-latest
+    if: |
+      (github.event_name == 'workflow_dispatch' || (github.event_name != 'workflow_dispatch' && github.repository == 'redis/redis'))
+    steps:
+    - name: prep
+      if: github.event_name == 'workflow_dispatch'
+      run: |
+        echo "GITHUB_REPOSITORY=${{github.event.inputs.use_repo}}" >> $GITHUB_ENV
+        echo "GITHUB_HEAD_REF=${{github.event.inputs.use_git_ref}}" >> $GITHUB_ENV
+    - uses: actions/checkout@v3
+      with:
+        repository: ${{ env.GITHUB_REPOSITORY }}
+        ref: ${{ env.GITHUB_HEAD_REF }}
+    - name: make
+      run: make REDIS_CFLAGS='-Werror'
+    - name: fadvise test
+      run: grep HAVE_FADVISE ./src/.make-settings
```

Comment:
i'm not certain i understand Yossi's original intent, but still i have a few comments about this:
1. we don't need a special job (or two) for this. it would be enough to add a step to an existing job (one on freebsd and one on linux).
2. maybe we don't need the freebsd one, and it's enough to know that on linux we detect it correctly, and see that other platforms don't fail to build/run.
3. maybe grepping the source code is not the right approach, and instead we can add a tcl test (that runs only on linux) which sets up some extreme scenario and then somehow measures the amount of page cache used by the process? i.e. validates that it achieves its purpose rather than just the fact we call fadvise.

Reply:
>  and instead we can add a tcl test (that runs only on linux) which sets up some extreme scenario and then somehow measures the amount of page cache used by the process?

Sadly there're no metrics about  page cache used by the process, `/proc/meminfo` counts the total cache size, but this cannot be used directly in tcl test, since the test environment is not isolated, other test case or processes may also increase the cache. The tool `vmtouch` can observe the cache size of file, but it need another dependency installation, so I won't try it. The syscall `mincore` can also observe the cache backed by a file, but I don't know how to use it in tcl... So I only add unittest for it now, it would be nice if you have other ideas about how to measure the cache in integration test.

Reply:
ok, so in the current approach we don't have any TCL tests that test our usage of fadvise (make sure we have no bugs).
but we do have a test (specific for linxu) that checks our detection and the fact fadvise works.

i agree we probably can't write a tcl test that verifies these, since we have no clue how these run and in parallel to what.
but maybe we can add a check in one of the GH action workflows?
i.e. in that case we do know we run on an isolated host.
e.g. spin up a job on linux, measure `/proc/meminfo` run `--single integration/replication` or alike, and re-measure meminfo?

Reply:
I added a github action but I'm not sure it works as expected.  There're some other variables, like restart loads the rdb, or log, that may increase the cache, so I doubt the limit 10KB is enough, and I don't know how to trigger a github action to verify it. I'll tune it according to the result.